pkg/node/node.go

```diff
@@ -278,6 +278,18 @@ func NewNode(
 		IntervalConfig:   nodeInfoPublisherInterval,
 	})
 
+	// Start periodic software update checks.
+	updateCheckCtx, stopUpdateChecks := context.WithCancel(ctx)
+	version.RunUpdateChecker(
+		updateCheckCtx,
+		func(ctx context.Context) (*models.BuildVersionInfo, error) { return nil, nil },
+		version.LogUpdateResponse,
+	)
+	config.CleanupManager.RegisterCallback(func() error {
```

Comment:
Probably okay for now, but I believe we're slowly moving away from the cleanup manager's. Can be fixed when the one below is migrated though.

Reply:
Yeah if we have defined what we're moving towards then I'm happy to do that instead. I haven't seen anything on that though.